up-l1/someip.adoc

```diff
@@ -0,0 +1,221 @@
+SOME/IP provides a framework for efficient and reliable communication between ECUs over IP networks. It is designed to support real-time and high-bandwidth communication requirements in automotive systems. The protocol allows ECUs to discover and communicate with each other, exchange messages, and provide services in a distributed system.
+SOME/IP is based on the Internet Protocol (IP) and uses UDP (User Datagram Protocol) as the transport layer. It supports various communication patterns, including unicast, multicast, and broadcast. The protocol defines a set of message formats and procedures for encoding and decoding data, handling service discovery, and managing communication sessions.
+By using SOME/IP, automotive ECUs can communicate with each other to exchange information related to vehicle functions, such as sensor data, control commands, diagnostics, and software updates. It enables the integration and interoperability of different ECUs from various automotive suppliers within a vehicle's network architecture.
+Overall, SOME/IP plays a crucial role in facilitating efficient and standardized communication between ECUs in modern automotive systems.
+
+The following specification will elaborate on how uProtocol shall be mapped to SOME/IP protocol so that mechatronics applications and services can be connected to non-mechatronics and cloud based applications and services. 
+
+=== Open Source Libraries
+
+As of the time of writing, there are two open source projects that are available for SOME/IP:
+
+1. *https://github.com/COVESA/vsomeip[COVESA-vsomeip]:* C++ solution based on the original specifications developed by BMW prior to contribution to AUTOSAR
+2. *https://projects.eclipse.org/projects/automotive.sommr[Eclipse-SommR]:*  Rust implementation based on the latest AUTOSAR SOME/IP specifications. 
+
+NOTE: Given the latest version of SOME/IP specifications from AUTOSAR is not open source, we cannot share or implement uProtocol using Eclipse-SommR.
+
+
+== Transport Layer Translation
+
+=== UAttributes
+
+The following table highlights the mapping of uAttributes to SOME/IP fields that apply to all event types. For more information about uAttributes please refer to link:../basics/uri.adoc[URI].
+
+
+==== Uris
+
+link:../basics/uattributes.adoc[UAttributes Specification] explains that `source` attributes defines the address of whom sent the message, while `sink` defines the destination for the message in link:../basics/uri.adoc[`UUri`] format. `UUri` is translated to SOME/IP attributes per  <<sink-source-uri-mapping>> below.
+
+.Sink & Source URI Mapping
+[sink-source-uri-mapping]
+|===
+|*UUri Portion* |*SOME/IP Field* |*Description*
+|*`UAuthority::ip`* |IP address & port |IP address & port (destination
+endpoint) of the mDevice
+
+|*`UEntity::id`* |Service ID |Software entity ID that is placed in the upper 16
+bits of the SOME/IP Message ID header field
+
```

Comment:
SOME/IP header does not include Instance ID, but it is used in SOME/IP SD and also required for vsomeip/sommr API.
Can we map it to UUri? e.g. ServiceID | InstanceID both u16 as `UAuthority::ip`: u32

[UUri spec](https://github.com/eclipse-uprotocol/up-spec/blob/someip-cleanup/basics/uri.adoc#uuri-specification) also refers to UResource instance, but maybe it is outdated.

Reply:
As I checked again UUri spec, we don't have u32 id to use for ServiceID + InstanceID.
It could be handled as a client option per Service ID along with major/minor service version.

Reply:
per discussion below, we will stick with instance 1. Need to discuss instance id further though.